javascript-carousel/main.js

```diff
@@ -0,0 +1,246 @@
+      viewAll[3].classList.remove('hidden');
+      viewAll[0].classList.add('hidden');
+      viewAll[1].classList.add('hidden');
+      viewAll[2].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+  });
+}
+
+var forwardButtonAll = document.querySelectorAll('.forward');
+for (var x = 0; x < forwardButtonAll.length; x++) {
+  forwardButtonAll[x].addEventListener('click', function (event) {
+    if (event.target.matches('.one')) {
+      tabAll[1].classList.add('active');
+      tabAll[0].classList.remove('active');
+      tabAll[2].classList.remove('active');
+      tabAll[3].classList.remove('active');
+      tabAll[4].classList.remove('active');
+      viewAll[1].classList.remove('hidden');
+      viewAll[0].classList.add('hidden');
+      viewAll[2].classList.add('hidden');
+      viewAll[3].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+
+    if (event.target.matches('.two')) {
+      tabAll[2].classList.add('active');
+      tabAll[0].classList.remove('active');
+      tabAll[1].classList.remove('active');
+      tabAll[3].classList.remove('active');
+      tabAll[4].classList.remove('active');
+      viewAll[2].classList.remove('hidden');
+      viewAll[0].classList.add('hidden');
+      viewAll[1].classList.add('hidden');
+      viewAll[3].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+
+    if (event.target.matches('.three')) {
+      tabAll[3].classList.add('active');
+      tabAll[0].classList.remove('active');
+      tabAll[1].classList.remove('active');
+      tabAll[2].classList.remove('active');
+      tabAll[4].classList.remove('active');
+      viewAll[3].classList.remove('hidden');
+      viewAll[0].classList.add('hidden');
+      viewAll[1].classList.add('hidden');
+      viewAll[2].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+
+    if (event.target.matches('.four')) {
+      tabAll[4].classList.add('active');
+      tabAll[0].classList.remove('active');
+      tabAll[1].classList.remove('active');
+      tabAll[2].classList.remove('active');
+      tabAll[3].classList.remove('active');
+      viewAll[4].classList.remove('hidden');
+      viewAll[0].classList.add('hidden');
+      viewAll[1].classList.add('hidden');
+      viewAll[2].classList.add('hidden');
+      viewAll[3].classList.add('hidden');
+    }
+
+    if (event.target.matches('.five')) {
+      tabAll[0].classList.add('active');
+      tabAll[1].classList.remove('active');
+      tabAll[2].classList.remove('active');
+      tabAll[3].classList.remove('active');
+      tabAll[4].classList.remove('active');
+      viewAll[0].classList.remove('hidden');
+      viewAll[1].classList.add('hidden');
+      viewAll[2].classList.add('hidden');
+      viewAll[3].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+  });
+}
+var libraryCounter = 0;
+var library = [
+  1, 2, 3, 4, 5
+];
+setInterval(changeSlide, 3000);
```

Comment:
The interval should restart every time the user interacts with the carousel controls.